### PR TITLE
use kustomize installed from makefile and remove unused install script

### LINKED ISSUE #491: Replace hack/ensure-*.sh scripts with `go install`
/kind feature

**Describe the solution you'd like**
In the `./hack` directory are scripts named `ensure-<something>.sh.` Some of these install Go-based tools, and they could perhaps be replaced entirely by invocations of `go install`.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR aligns the CI and install shell scripts in the repo to use kustomize binary installed in hack/tools/bin/ directory.
Shell scripts also check/installs missing kustomize tools in hack/tools/bin/ dir, via Makefile recipes, before using it in the shell script.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #491 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
use kustomize installed from makefile and remove unused install script
```

### CODE REVIEW

**Comment by sayantani11** [comment]
@cpanato Can you please help with the error? I am still not able to get through what I am missing.

**Comment by evanfreed** [comment]
@sayantani11 I think you might need something like this in your makefile.  Its a bit confusing 

```
.PHONY: $(KUSTOMIZE_BIN)
$(KUSTOMIZE_BIN): $(KUSTOMIZE) ## Building Kustomize from tools folder

OR 

kustomize: $(KUSTOMIZE) ## Build a local copy of kustomize.
```

The first one mimics what other targets in the GCP Makefile follow.

**Comment by sayantani11** [comment]
@evanfreed I actually was thinking the same thing. But I wasn't sure if I
should, but will give it a try.

**Comment by sayantani11** [comment]
@cpanato I am not able to understand the test-e2e error, although I tried fixing it but I am not able to understand why the error is still there.

File: Makefile [hunk 1]
```diff
@@ -275,6 +275,9 @@ $(KIND): ## Build kind into tools folder
 .PHONY: $(KUBECTL_BIN)
 $(KUBECTL_BIN): $(KUBECTL) ## Building kubectl from tools folder
 
+.PHONY: $(KUSTOMIZE_BIN)
+	$(KUSTOMIZE_BIN): $(KUSTOMIZE) ## Building Kustomize from tools folder
```
Comment (cpanato) [inline Makefile:279]: maybe need to make it similar to the others, and remove the indentation?

Comment (sayantani11) [reply Makefile:279]: Ok I see... Let me try fixing it

**Comment by sayantani11** [comment]
cc @cpanato

**Comment by cpanato** [comment]
you might need to rebase to catch new things from the main branch

**Pull request closed without merging**